plugins/pulp_docker/plugins/importers/sync.py

```diff
@@ -187,122 +130,62 @@ def process_main(self):
-                                            model.relative_path)
-
+        super(SaveUnitsStep, self).__init__(step_type=constants.SYNC_STEP_SAVE)
+        self.description = _('Saving Manifests and Blobs')
 
-class SaveUnitsStep(PluginStep):
-    def __init__(self, working_dir):
+    def get_iterator(self):
         """
-        :param working_dir: full path to the directory into which blob files
-                            are downloaded. This directory should contain one
-                            directory for each docker blob, with the ID of the
-                            docker blob as its name.
-        :type  working_dir: basestring
+        Return an iterator that will traverse the list of Units that were downloaded.
+
+        :return: An iterable containing the Blobs and Manifests that were downloaded and are new to
+                 Pulp.
+        :rtype:  iterator
         """
-        super(SaveUnitsStep, self).__init__(
-            step_type=constants.SYNC_STEP_SAVE, plugin_type=constants.IMPORTER_TYPE_ID,
-            working_dir=working_dir)
-        self.description = _('Saving manifests and blobs')
+        return iter(self.parent.step_get_local_manifests.units_to_download +
+                    self.parent.step_get_local_blobs.units_to_download)
 
-    def process_main(self):
+    def process_main(self, item=None):
         """
-        Gets an iterable of units that were downloaded from the parent step,
+        Gets an iterable of Units that were downloaded from the parent step,
         moves their files into permanent storage, and then saves the unit into
         the database and into the repository.
-        """
-        _logger.debug(self.description)
-        # Save the Manifests
-        for unit_key in self.parent.step_get_metadata.step_get_local_units.units_to_download:
-            model = self.parent.step_get_metadata.manifests[unit_key['digest']]
-            unit = self.get_conduit().init_unit(model.TYPE_ID, model.unit_key, model.metadata,
-                                                model.relative_path)
-            self._move_file(unit)
-            _logger.debug('saving manifest %s' % model.digest)
-            self.get_conduit().save_unit(unit)
 
-        # Save the Blobs
-        for unit_key in self.parent.step_get_local_units.units_to_download:
-            model = models.Blob(unit_key['digest'])
-            unit = self.get_conduit().init_unit(model.TYPE_ID, model.unit_key, model.metadata,
-                                                model.relative_path)
-            self._move_file(unit)
-            _logger.debug('saving Blob %s' % unit_key)
-            self.get_conduit().save_unit(unit)
-
-    def _move_file(self, unit):
-        """
-        For the given unit, move its associated file from the working
-        directory to its permanent location.
+        This method will be called exactly once without an item argument if the iterator returned by
```

Comment:
I actually fixed this recently. https://github.com/pulp/pulp/pull/2185/files#diff-2df6878623b5e8e58165b6d4142234e1R224

I think it was more a simple coding mistake than an anti-pattern. It was the common mistake of depending on the automatic evaluation ``bool(thing)`` for the expression ``thing`` when they really should have used ``thing is not None``.

Explaining why you had to do something unusual is valuable, so thanks for doing that. It helped us connect on the fact that the underlying problem was fixed. But while I know you're not a fan of that code in general, going on to express your opinion about it does not add value, and I don't think this is the right place to do so.

In any case, given that this has been fixed, you can remove the ``if`` statement.